File: Client-app/src/Modules/Employee/Employee.Domain/employee.ts
export interface Employee {
  id?: number;
  fullName?: string;
  email?: string;
  phone?: string;
  hireDate?: string;
  salary?: number;
  department?: Department;
  isActive?: boolean;
}

export enum Department {
  HR = 0,
  IT = 1,
  Finance = 2,
}

File: Domain/Entities/Employee.cs
using System;
using EmployeeManagement.Domain.Enums;

namespace EmployeeManagement.Domain.Entities
{
    public class Employee
    {
        public int        Id         { get; set; }
        public string     FullName   { get; set; }
        public string     Email      { get; set; }
        public string     Phone      { get; set; }
        public DateTime   HireDate   { get; set; }
        public decimal    Salary     { get; set; }
        public Department Department { get; set; }
        public bool       IsActive   { get; set; } = true;
        public bool       IsDeleted  { get; set; } = false;
    }
}
